cpp/include/cudf/strings/strings_column_handler.hpp

```diff
@@ -0,0 +1,232 @@
+   *---------------------------------------------------------------------------**/
+  column_view chars_column() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns a pointer to the internal null mask memory
+   *---------------------------------------------------------------------------**/
+  const bitmask_type* null_mask() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the number of nulls in this column
+   *---------------------------------------------------------------------------**/
+  size_type null_count() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the registered memory resource
+   *---------------------------------------------------------------------------**/
+  rmm::mr::device_memory_resource* memory_resource() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Prints the strings to stdout.
+   *
+   * @param start Index of first string to print.
+   * @param end Index of last string to print. Specify -1 for all strings.
+   * @param max_width Maximum number of characters to print per string.
+   *        Specify -1 to print all characters.
+   * @param delimiter The chars to print between each string.
+   *        Default is new-line character.
+   *---------------------------------------------------------------------------**/
+  void print( size_type start=0, size_type end=-1,
+              size_type max_width=-1, const char* delimiter = "\n" ) const;
+
+  // sort types can be combined
+  enum sort_type {
+      none=0,    ///< no sorting
+      length=1,  ///< sort by string length
+      name=2     ///< sort by characters code-points
+  };
+
+private:
+  const column_view _parent;
+  rmm::mr::device_memory_resource* _mr;
+
+};
+
+namespace strings
+{
+
+// array.cu
+/**---------------------------------------------------------------------------*
+ * @brief Returns a new strings column created from a subset of
+ * of this instance's strings column.
+ *
+ * @code
+ * s1 = ["a", "b", "c", "d", "e", "f"]
+ * s2 = sublist( s1, 2 )
+ * s2 is ["c", "d", "e", "f"]
+ * @endcode
+ * 
+ * @param start Index of first string to use.
+ * @param end Index of last string to use.
+ *        Default -1 indicates the last element.
+ * @param step Increment value between indexes.
+ *        Default step is 1.
+ * @param stream CUDA stream to use kernels in this method.
+ * @return New strings column of size (end-start)/step.
+ *---------------------------------------------------------------------------**/
+std::unique_ptr<cudf::column> sublist( strings_column_handler handler,
+                                       size_type start, size_type end=-1,
+                                       size_type step=1,
+                                       cudaStream_t stream=(cudaStream_t)0 );
```

Comment:
There's no need to cast this. 
```suggestion
                                       cudaStream_t stream=0 );
```